src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -225,13 +225,13 @@ protected internal virtual Task<Stream> ReadJSDataAsStreamAsync(IJSStreamReferen
     }
 
     [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:RequiresUnreferencedCode", Justification = "We enforce trimmer attributes for JSON deserialized types on InvokeAsync.")]
-    internal void EndInvokeJS(long taskId, bool succeeded, ref Utf8JsonReader jsonReader)
+    internal bool EndInvokeJS(long taskId, bool succeeded, ref Utf8JsonReader jsonReader)
     {
         if (!_pendingTasks.TryRemove(taskId, out var tcs))
         {
             // We should simply return if we can't find an id for the invocation.
             // This likely means that the method that initiated the call defined a timeout and stopped waiting.
-            return;
+            return false;
```

Comment:
Could you remind me what the workflow looks like in this case? The EndInvokeJS is called by JS but the task on the server is removed, so there's nothing to do?

Reply:
Exactly. So for WASM this would look something like 

Call Tree:
- JS
- DefaultWebAssemblyJSRuntime.EndInvokeJS
- DotNetDispatcher.EndInvokeJS
- JSRuntime.EndInvokeJS (here)

Reply:
Ok. As long as nothing (incidentally) relied on the JSException being previously thrown, this looks great.

Reply:
Nothing that I can tell. This is also the only call to `JSRuntime.EndInvokeJS`: https://source.dot.net/#Microsoft.JSInterop/JSRuntime.cs,64701806bdd192b1,references